src/naf-dialog-adapter.js

```diff
@@ -519,9 +442,55 @@ export default class DialogAdapter {
     await Promise.all([this.updateTimeOffset(), this._initialAudioConsumerPromise]);
   }
 
+  newPeer(peer) {
+    this._onOccupantConnected(peer.id);
+    this.occupants[peer.id] = peer;
+
+    if (this._onOccupantsChanged) {
+      this._onOccupantsChanged(this.occupants);
+    }
+  }
+
+  closePeer(peerId) {
+    this._onOccupantDisconnected(peerId);
+
+    const pendingMediaRequests = this._pendingMediaRequests.get(peerId);
+
+    if (pendingMediaRequests) {
+      const msg = "The user disconnected before the media stream was resolved.";
+      info(msg);
+
+      if (pendingMediaRequests.audio) {
+        pendingMediaRequests.audio.resolve(null);
+      }
+
+      if (pendingMediaRequests.video) {
+        pendingMediaRequests.video.resolve(null);
+      }
+
+      this._pendingMediaRequests.delete(peerId);
+    }
+
+    // Resolve initial audio resolver since this person left.
+    const initialAudioResolver = this._initialAudioConsumerResolvers.get(peerId);
```

Comment:
Not part of this PR but the whole `_initialAudioConsumerResolvers` thing is something I want to look into soon. This stuff still feels a bit fishy to be and would like to avoid it if we can. I would bet there are cases where we end up stalling on this for whatever reason. Its unclear why we need to wait for this.

Reply:
I guess that covers the case where that peer disconnected before it's audio consumer resolved which is something that we do upon connect to wait until we receive all the other peer's consumers. I agree that can cause some issues. I'm not sure why we do that though, is it really that important that we wait until we can hear all the other peers to connect?